### PR TITLE
Add encoding utils

### PR DESCRIPTION
**Pull request by HzjNeverStop**

### CODE REVIEW

**Comment by OrezzerO** [comment]
I don't think it is a good idea to check if a byte array is UTF-8 encoded. It is more reasonable to test if default charset is UTF-8.

File: src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+/**
+ * Util to determine Charsets, the implements fork from guava.
+ *
+ * @author huzijie
+ * @version CharsetUtil.java, v 0.1 2023年04月14日 2:06 PM huzijie Exp $
+ * @see com.google.common.base.Utf8#isWellFormed(byte[])
+ */
+public class CharsetUtil {
+
+    public static Logger CHARSET_MONITOR_LOG = LoggerFactory.getLogger("CHARSET_MONITOR_LOG");
```
Comment (ujjboy) [inline src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java:33]: 这个配置方式是希望用户自己配 CHARSET_MONITOR_LOG 吗？

File: src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+    /**
+     * Asserts the given byte array is well formed in UTF-8 encoding format.
+     * @param bytes The byte array to be checked.
+     */
+    public static void assertUtf8WellFormed(byte[] bytes) {
+        checkUtf8WellFormed(bytes, 0);
```
Comment (ujjboy) [inline src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java:40]: 0  换成常量吧

File: src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    public static void assertUtf8WellFormed(byte[] bytes, int off, int len) {
+        checkUtf8WellFormed(bytes, off, len, 0);
+    }
+
+    /**
+     * Monitor the given byte array is well formed in UTF-8 encoding format.
+     * @param bytes The byte array to be checked.
+     */
+    public static void monitorUtf8WellFormed(byte[] bytes) {
+        checkUtf8WellFormed(bytes, 1);
```
Comment (ujjboy) [inline src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java:58]: 1 换成常量

File: src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+ */
+public class CharsetUtil {
+
+    public static Logger CHARSET_MONITOR_LOG = LoggerFactory.getLogger("CHARSET_MONITOR_LOG");
+
+    /**
+     * Asserts the given byte array is well formed in UTF-8 encoding format.
+     * @param bytes The byte array to be checked.
+     */
+    public static void assertUtf8WellFormed(byte[] bytes) {
```
Comment (ujjboy) [inline src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java:39]: utf-8  里的  UTF 全部大写吧，这个是指定缩写

**Review by ujjboy — CHANGES_REQUESTED** [review]

**Comment by HzjNeverStop** [comment]
> I don't think it is a good idea to check if a byte array is UTF-8 encoded. It is more reasonable to test if default charset is UTF-8.

This utility class is used to detect whether a system that uses UTF-8 encoding has read non-UTF-8 encoded data from other data streams.

File: src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java (hunk 1 above)
Comment (HzjNeverStop) [reply src/main/java/com/alipay/sofa/common/utils/CharsetUtil.java:33]: > 这个配置方式是希望用户自己配 CHARSET_MONITOR_LOG 吗？

是的，用户自己配置 LOGGER 更加灵活，例如可以为日志打印添加MDC（traceId、方法入参）

**Review by ujjboy — APPROVED** [review]
LGTM

**Pull request merged**